crates/oxc_linter/src/rules/eslint/no_misleading_character_class.rs

```diff
@@ -0,0 +1,654 @@
+        ("var r = RegExp(/[👍]/u)", None),
+        ("const regex = /[👍]/u; new RegExp(regex);", None),
+        // ("new RegExp('[👍]')", None), // { "globals": { "RegExp": "off" } },
+        // Ignore solo lead/tail surrogate.
+        (r"var r = /[\uD83D]/", None),
+        (r"var r = /[\uDC4D]/", None),
+        (r"var r = /[\uD83D]/u", None),
+        (r"var r = /[\uDC4D]/u", None),
+        // Ignore solo combining char.
+        (r"var r = /[\u0301]/", None),
+        (r"var r = /[\uFE0F]/", None),
+        (r"var r = /[\u0301]/u", None),
+        (r"var r = /[\uFE0F]/u", None),
+        // Ignore solo emoji modifier.
+        (r"var r = /[\u{1F3FB}]/u", None),
+        ("var r = /[🏻]/u", None),
+        // Ignore solo regional indicator symbol.
+        ("var r = /[🇯]/u", None),
+        ("var r = /[🇵]/u", None),
+        // Ignore solo ZWJ.
+        (r"var r = /[\u200D]/", None),
+        (r"var r = /[\u200D]/u", None),
+        // don't report and don't crash on invalid regex
+        ("new RegExp('[Á] [ ');", None),
+        ("var r = new RegExp('[Á] [ ');", None),
+        ("var r = RegExp('{ [Á]', 'u');", None),
+        ("var r = new globalThis.RegExp('[Á] [ ');", None), // { "ecmaVersion": 2020 },
+        ("var r = globalThis.RegExp('{ [Á]', 'u');", None), // { "ecmaVersion": 2020 },
+        // don't report on templates with expressions
+        ("var r = RegExp(`${x}[👍]`)", None),
+        // don't report on unknown flags
+        ("var r = new RegExp('[🇯🇵]', `${foo}`)", None),
+        // (r#"var r = new RegExp("[👍]", flags)"#, None),
+        // don't report on spread arguments
+        ("const args = ['[👍]', 'i']; new RegExp(...args);", None),
+        ("var r = /[👍]/v", None),         // { "ecmaVersion": 2024 },
+        (r"var r = /^[\q{👶🏻}]$/v", None),  // { "ecmaVersion": 2024 },
+        (r"var r = /[🇯\q{abc}🇵]/v", None), // { "ecmaVersion": 2024 },
+        ("var r = /[🇯[A]🇵]/v", None),      // { "ecmaVersion": 2024 },
+        ("var r = /[🇯[A--B]🇵]/v", None),   // { "ecmaVersion": 2024 },
+        (r"/[\ud83d\udc4d]/", Some(serde_json::json!([{ "allowEscape": true }]))),
+        (
+            r#"/[�d83d\udc4d]/u // U+D83D + Backslash + "udc4d""#,
+            Some(serde_json::json!([{ "allowEscape": true }])),
+        ),
+        (r"/[A\u0301]/", Some(serde_json::json!([{ "allowEscape": true }]))),
+        (r"/[👶\u{1f3fb}]/u", Some(serde_json::json!([{ "allowEscape": true }]))),
+        (r"/[\u{1F1EF}\u{1F1F5}]/u", Some(serde_json::json!([{ "allowEscape": true }]))),
+        (r"/[👨\u200d👩\u200d👦]/u", Some(serde_json::json!([{ "allowEscape": true }]))),
+        (r"/[\u00B7\u0300-\u036F]/u", Some(serde_json::json!([{ "allowEscape": true }]))),
+        (r"/[\n\u0305]/", Some(serde_json::json!([{ "allowEscape": true }]))),
+        // (r#"RegExp("[\uD83D\uDC4D]")"#, Some(serde_json::json!([{ "allowEscape": true }]))),
+        // (r#"RegExp("[A\u0301]")"#, Some(serde_json::json!([{ "allowEscape": true }]))),
```

Comment:
@leaysgur I expect that this should work because it works with the `RegExpLiteral` in line 448.
The `Character.kind` is `CharacterKind::Symbol` instead of `UnicodeEscape`.

Reply:
https://github.com/oxc-project/oxc/issues/13660#issue-3403241660